File: anisearchEx/searchEntry.py
from anisearchEx.anisearchLoader import load
import base64

BASE_LINK = "https://cdn.anisearch.com/images/"
class SearchEntry:
    def __init__(self, name, imageLink, link) -> None:
        self.name = name
        self.image = self._downloadImage(self.__convertToImageLink(imageLink))
        self.link = link


    def __convertToImageLink(self, link):
        bigImageLink = link.replace("full/", "")
        bigImageLink = bigImageLink.replace(".webp", "_300.webp")
        return f"{BASE_LINK}{bigImageLink}"

    
    def _downloadImage(self, link) -> str:        
        base64Image = base64.b64encode(load(link)).decode()
        return f"data:image/webp;base64,{base64Image}"

File: src/anisearchEx/searchEntry.ts
// import base64 from 'base-64';
import {loadImageAsBase64} from './anisearchLoader';

const BASE_LINK = "https://cdn.anisearch.com/images/";

export class SearchEntry {
    name:string;
    imageLink:string;
    link:string;
    image:string;

    constructor(name:string, imageLink:string, link:string) {
        this.name = name;
        this.imageLink = imageLink;
        this.link = link;
        this.image = "";
    }

    #convertToImageLink(link:string) {
        return `${BASE_LINK}${link}`;
    }

    async loadImage() {
        let base64Image = (await loadImageAsBase64(this.#convertToImageLink(this.imageLink))).message;
        this.image = `data:image/webp;base64,${base64Image}`;
    }
}